stdlib/src/builtin/string.mojo

```diff
@@ -930,12 +930,13 @@ struct String(
         Returns:
             A new string containing the character at the specified position.
         """
+        var index = idx
         if idx < 0:
-            return self.__getitem__(len(self) + idx)
+            index = len(self) + index
```

Comment:
Thank you for the pull request, indeed this needed to be fixed! The is a function to do this operation. Can you take a look at `normalize_index`? See https://github.com/modularml/mojo/issues/2948 for more information on the subject :)

Reply:
I'll have a look, thank you!